src/components/Contractor/Profile/useContractorProfile.ts

```diff
@@ -0,0 +1,357 @@
+        middleInitial: data.middleInitial || undefined,
+        lastName: data.lastName,
+        ssn: data.ssn,
+        fileNewHireReport: false, // Default value
+      }
+    } else {
+      return {
+        ...basePayload,
+        businessName: data.businessName,
+        ein: data.ein,
+      }
+    }
+  }
+
+  const transformFormDataToUpdatePayload = (
+    data: ContractorProfileFormData,
+    version: string,
+  ): PutV1ContractorsContractorUuidRequestBody => {
+    const createPayload = transformFormDataToCreatePayload(data)
+    return {
+      ...createPayload,
+      version,
+    }
+  }
+
+  // Event handlers
+  const onSubmit: SubmitHandler<ContractorProfileFormData> = async data => {
+    await baseSubmitHandler(data, async payload => {
+      if (contractorId && existingContractor) {
+        // Update existing contractor
+        if (!existingContractor.version) {
+          throw new Error('Contractor version is required for updates')
+        }
+        const version = String(existingContractor.version)
+        const apiPayload = transformFormDataToUpdatePayload(payload, version)
+
+        const updateResponse = await updateContractor({
+          request: {
+            contractorUuid: contractorId,
+            requestBody: apiPayload,
+          },
+        })
+
+        onEvent(componentEvents.CONTRACTOR_UPDATED, updateResponse.contractor)
+      } else {
+        // Create new contractor
+        const apiPayload = transformFormDataToCreatePayload(payload)
+
+        const createResponse = await createContractor({
+          request: {
+            companyUuid: companyId,
+            requestBody: apiPayload,
+          },
+        })
+
+        onEvent(componentEvents.CONTRACTOR_CREATED, createResponse.contractor)
+      }
+
+      onEvent(componentEvents.CONTRACTOR_PROFILE_SUBMITTED, payload)
+    })
+  }
+
+  const handleCancel = () => {
+    onEvent(componentEvents.CANCEL)
+  }
+
+  // Conditional rendering helpers
+  const shouldShowEmailField = watchedInviteContractor
+  const shouldShowBusinessFields = watchedType === ContractorType.Business
+  const shouldShowIndividualFields = watchedType === ContractorType.Individual
+  const shouldShowHourlyRate = watchedWageType === WageType.Hourly
+
+  // Form field options
+  const contractorTypeOptions = [
+    { label: 'Individual', value: ContractorType.Individual },
+    { label: 'Business', value: ContractorType.Business },
+  ]
+
+  const wageTypeOptions = [
+    { label: 'Hourly', value: WageType.Hourly },
+    { label: 'Fixed', value: WageType.Fixed },
+  ]
+
+  // Determine if we're currently submitting (creating or updating)
+  const isSubmitting = isCreating || isUpdating
+
+  return {
+    // Form methods
+    formMethods,
+    handleSubmit: handleSubmit(onSubmit),
+    formState: {
+      ...formState,
+      isSubmitting,
+    },
+
+    // Watched values
+    watchedType,
+    watchedWageType,
+    watchedInviteContractor,
```

Comment:
Nit: Thinking through this API i'm feeling complicated about passing through all the things with hook form as a return. wondering if there's a way to only surface what's actually needed by the parent component? maybe it's just the idea that we're hard coding hook form stuffs as part of our api contracts for both the hook and the ui component that's throwing me off

Reply:
Updated